webpack/assets/javascripts/react_app/components/PermissionDenied/index.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Button, Icon } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../common/I18n';
+
+const PermissionDenied = ({ missingPermissions }) => {
+  const permissionsList = missingPermissions.map(permission => (
+    <li key={permission}>
+      <strong>{permission}</strong>
+    </li>
+  ));
+
+  return (
+    <div className="blank-slate-pf">
+      <div className="blank-slate-pf-icon">
+        <Icon name="lock" color="#9c9c9c" size="2x" title="lock" />
+      </div>
+      {__('You are not authorized to perform this action.')}
+
+      <h1>{__('Permission denied')}</h1>
+      <p>
+        {__('You are not authorized to perform this action.')}
+        <br />
+        {__(
+          'Please request one of the required permissions listed below from a Foreman administrator:'
+        )}
+        <br />
+      </p>
+      <ul className="list-unstyled">{permissionsList}</ul>
+      <p>
+        <a href="/">
```

Comment:
I would like to see this configurable (having an option).
As in rails we would likely pass `main_app.root_path` what doesn't always need to resolve in `'/'`.

Reply:
+1 @ezr-ondrej - We might want to use the same mechanism with react-router.

I am also wondering if we should use the react-router `Link` component instead of a regular `a` tag.

Reply:
@sharvit `<Link />` just uses the history API and doesn't cause the browser to do a GET request, will that work if we are linking to a non-React page?  I think I've had problems like that with one of my bugfix PRs.

Reply:
I see your point.

Can we make it work for both cases?
Phrases by something like `{ clientSideRoute ? <Link /> : <a /> }`?

Another option, use the recived `linkTo` prop (or whatever you decided to call it) and try figure out if it's a `clientSideRoute` or not. @amirfefer do you think it is reasonable?

Reply:
@sharvit I like the second idea!  But Foreman seems to have very odd ways of storing the routes (see like `webpack/assets/javascripts/react_app/routes/routes.js`), so I'm not sure of the best way to check them.  Hopefully you're fixing this in your client side routing project?

For now I built a `<ForemanLink />` component which is kind of in between the two ideas.. see what you think.

Reply:
@amirfefer working on finalizing the client routes so he might have a clue.

+1 for creating `ForemanLink`, I think it is going to be a key part in the foreman infrastructure.

Reply:
I have been talking with @amirfefer about `ForemanLink` and we figure out it should be possible to use the normal `Link` since we have a special fallback route that fallback to use turbolinks when no other route is found.

Reply:
So the direction should be using `Link` and wrap the component with `withRouter`?
If so, should it be done in the `index.js` or in the component file? @amirfefer ?